src/Symfony/Component/Form/ChoiceList/Factory/Cache/AbstractStaticCallback.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/*
+ * This file is part of the BlackfireBundle project.
+ *
+ * (c) Jules Pietri <jules@heahprod.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\ChoiceList\Factory\Cache;
+
+use Symfony\Component\Form\ChoiceList\Factory\CachingFactoryDecorator;
+use Symfony\Component\Form\FormTypeInterface;
+
+/**
+ * A template for static callbacks used by {@see CachingFactoryDecorator}.
+ *
+ * @internal
+ *
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+abstract class AbstractStaticCallback
+{
+    private static $callbacks = [];
+
+    /** @var callable|string|array|\Closure */
+    private $callback;
+
+    /**
+     * @param FormTypeInterface $formType A form type configuring a list
+     * @param callable          $callback Any pseudo callable to create choice list property
+     * @param mixed             $vary     Dynamic data used to compute a unique hash when caching the callback
+     */
+    final public function __construct(FormTypeInterface $formType, $callback, $vary = null)
```

Comment:
Why not
```suggestion
    final public function __construct(FormTypeInterface $formType, callable $callback, $vary = null)
```
?

Reply:
Saw the comment above to late, sorry.

Reply:
Let's then be explicit about it? `callable|string Any callable or property path to create choice list property`

Reply:
Actually we don't know what decorator will be used here, each decorator provides its own consistent phpdocs, but those properties are never type hinted to be used in factories methods.
Furthermore, userland and third party code can also decorate the factory.